app/src/pages/admin/allUsersPage/usersToolbar/actionPanel/actionPanel.jsx

```diff
@@ -24,16 +31,72 @@ const messages = defineMessages({
     id: 'AllUsersPage.AddUser',
     defaultMessage: 'Add user',
   },
+  addUserNotification: {
+    id: 'ActionPanel.addUserNotification',
+    defaultMessage: 'Member was successfully created',
+  },
+  addUserTitle: {
+    id: 'ActionPanel.addUserTitle',
+    defaultMessage: 'Add user',
+  },
+  submitText: { id: 'ActionPanel.submitText', defaultMessage: 'Add' },
+  cancelText: { id: 'ActionPanel.cancelText', defaultMessage: 'Cancel' },
 });
 
+@connect(
+  (state) => ({
+    users: allUsersSelector(state),
+  }),
+  {
+    showModal: showModalAction,
+    showNotification,
+  },
+)
 @injectIntl
 export class ActionPanel extends Component {
   static propTypes = {
     intl: intlShape.isRequired,
+    showNotification: PropTypes.func.isRequired,
+    showModal: PropTypes.func,
+  };
+  static defaultProps = {
+    showModal: () => {},
   };
   onExportUsers = () => {
     window.location.href = URLS.exportUsers();
   };
+  onAddUser = () => {
+    const { intl, showModal } = this.props;
+    const onSubmit = (values) => {
```

Comment:
Instead of creating onSubmit every time, please move inner logic to a new method `addUser`